### ISSUE TITLE
Include meta-programming capabilities?

### ISSUE THREAD

**Issue by kaneplusplus** [matloff/R-vs.-Python-for-Data-Science#5]
I think this is a great comparison that maintains a level of objectivity we don't usually see with these types of comparisons.

Would it be worth including meta-programming facilities? I think R is far and away the winner and I think it's one of the core reasons R is able to facilitate the construction of DSLs (like a large portion of the tidyverse) in ways that are currently not possible in Python.

**Comment by matloff** [comment]
I did mention metaprog. It might be interesting to add a comment re connection to the tidyverse.

**Comment by kaneplusplus** [comment]
Sorry, this was an oversight on my part.

The MP-tidyverse connection might help people understand the reason tidyverse exists is because of the flexibility designed into R. Practically, the "base vs. tidyverse" debate is about programming style. Technically, it is a false dichotomy since it implies that you can have the tidyverse without R's core facilities.

**Comment by matloff** [comment]
Yes, I suspect that some of those blindly praising the Tidyverse don't realize it's written in base R. :-)
